src/app/modules/design-patterns/adapter/interfaces/xml-to-json-adapter.ts

```diff
@@ -0,0 +1,24 @@
+import { Observable, throwError } from 'rxjs';
+import { xml2json } from 'xml-js';
+
+import { IAdapter } from './adapter';
+
+export class XMLtoJsonAdapter implements IAdapter {
+  constructor(private readonly xml: string) {
+    this.convert();
+  }
+
+  convert(): string {
+    try {
+      const json = xml2json(this.xml, {compact: true, spaces: 4});
+
+      return json;
+    } catch (error) {
+      this.invalidXMLException(error);
+    }
+  }
+
+  invalidXMLException(error: string): Observable<string> {
```

Comment:
This is not needed anymore since I asked u previously to throw exception directly